src/main.js

```diff
@@ -65,11 +65,25 @@ const load = loader({
     },
   },
 
+  pulseClient: {
+    requires: ['cfg', 'monitor'],
+    setup: ({cfg, monitor}) => {
+      return new libPulse.Client({
+        namespace: 'taskcluster-auth',
+        monitor,
+        credentials: libPulse.pulseCredentials(cfg.pulse),
+      });
+    },
+  },
+
   publisher: {
     requires: ['cfg', 'schemaset', 'monitor'],
-    setup: async ({cfg, schemaset, monitor}) => exchanges.setup({
+    setup: async ({cfg, schemaset, monitor}) => await exchanges.publisher({
       rootUrl:            cfg.taskcluster.rootUrl,
+      client:             pulseClient,
```

Comment:
This is the undefined reference.  `pulseClient` should be an argument to the setup function, similar to `monitor` a few lines above.  And if it's an argument, then it needs to be included in `requires` (again just like `'monitor'`).